cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
+Document directives instruct the browser about the properties of the document to which the policies will apply to. 
 
-CSP Sample Policies
-===================
+- `base-uri` specifies the possible URLs that the `<base>` element can use.
+- `plugin-types` limits the types of resources that can be loaded into the document (*e.g.* application/pdf). 3 rules apply to the affected elements, `<embed>` and `<object>`:
+  - The element needs to explicitly declare its type.
+  - The element's type needs to match the declared type.
+  - The element's resource need to meatch the declared type.
+- `sandbox` restricts a page's actions such as submitting forms.
+  - Only applies when used with the request header `Content-Security-Policy`.
+  - Not specifying a value for the directive activates all of the sandbox restrictions. `Content-Security-Policy: sandbox;`
+  - [Sandbox syntax](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/sandbox#Syntax)
 
-Basic CSP Policy
-----------------
+## Navigation Directives
 
-This policy will only allow resources from the originating domain for all the default level directives, and will not allow inline scripts/styles to execute. If your application and function with these restrictions, it drastically reduces your attack surface having this policy in place, and will work with most modern browsers.
+Navigation directives instruct the browser about the locations that the document can navigate to.
 
-The most basic policy assumes:
+- `navigate-to` restricts the URLs which a document can navigate to by any mean.
+- `form-action` restricts the URLs which the forms can submit to.
+- `frame-ancestors` restricts the URLs that can embed the requested resource inside of  `<frame>`, `<iframe>`, `<object>`, `<embed>`, or `<applet>` elements.
+  - If this directive is specified in a `<meta>` tag, the directive is ignored.
+  - This directive doesn't fallback to `default-src` directive.
+  - `X-Frame-Options` is rendered obsolete by this directive and is ignored by the user agents.
 
--   all resources are hosted by the same domain of the document
--   there are no inlines or evals for scripts and style resources
+## Reporting Directives
 
-` Content-Security-Policy: default-src 'self' `
+Reporting directives deliver violation of prevented behaviors to specified locations. These directives serve no purpose on their own and are dependent on other directives.
 
-To tighten further, one can do the following:
+- `report-to` which is a groupname defined in the header in a json formatted header value. Does not have proper browser support yet.
+  - [MDN report-to documentation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-to#Examples)
+- `report-uri` directive will be getting deprected by `report-to`, which is a URI that the reports are sent to.
+  - Goes by the format of: `Content-Security-Policy: report-uri https://example.com/csp-reports`
```

Comment:
Usage [example](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-uri).

Reply:
I'll be linking the above mentioned document. I believe writing it out is a bit too much. What do you think?

Reply:
Link is enough 👍  so it allow the CS reader to explore this point if needed.